rfcs/0000-rfc-forms.md

```diff
@@ -0,0 +1,148 @@
+---
+Status: Active
+Champions: christian24
+PR: {{ update_with_pr_number }}
+---
+
+# @lit-labs/forms
+
+Add advanced form validation for web components. This is a followup to the discussion in the lit repository: https://github.com/lit/lit/discussions/2489 
+
+## Objective
+
+Provide an easy to use dev experience for validating forms with web components in them. 
+
+### Goals
+- Provide a common API, so developers can quickly start using the @lit-labs/forms package without changing their existing controls. 
+- Allow cross field validation
+- Allow async validation
+- get and set dirty, touched, valid states for individual controls
+- allow for configuration (e. g. change on `blur` or `change`)
+- It should be typed
+- It should be possible to set initial values for individual controls. 
+- subscribe to value changes and update dependant controls accordinly. 
```

Comment:
AFAIK there's no concept of dependent control in this scenario (eg. a control which is automatically recalculated when another control changes), but you could subscribe to a control and imperatively mutate the other one.

In my prototype I could do that thanks to RxJS but it could be done via simple callbacks perhaps.

Reply:
I tried making this more clear: as  @UserGalileo said there is no concept of a dependent control, but you may have logic when one field changes, this automatically changes another field as well. 

This is quite interesting in practice to be honest. At first we used callbacks for this, which worked, but then when we turned it into a multi step wizard and we had to repopulate the form from the store, this would make changes we were not anticipating, since the input was already there, there was no need to trigger any of that logic. So we split this into two places: We used event listeners bindings directly in the template to detect if the user had changed something, because only then we want to update fields. Secondly, some values might also have an effect on rendering logic: Think about a combo box where you select a certain value and another text field appears, because additional input is needed. In those cases we would have derive the `@state()` from form values when changing wizard steps.